src/main/java/de/tum/in/www1/artemis/service/exam/ExamService.java

```diff
@@ -448,7 +448,6 @@ public void filterParticipationForExercise(StudentExam studentExam, Exercise exe
         if (!isAtLeastInstructor) {
             // If the exerciseGroup (and the exam) will be filtered out, move example solution publication date to the exercise to preserve this information.
             exercise.setExampleSolutionPublicationDate(exercise.getExerciseGroup().getExam().getExampleSolutionPublicationDate());
-            exercise.setExerciseGroup(null);
```

Comment:
we might need to set exerciseGroup.core to null to avoid making the json too large

Reply:
what do you mean with exerciseGroup.core?
exerciseGroup.exercises? or exerciseGroup.exam or both?

Reply:
sorry, it was a typo. I actually meant setting the exam to null